core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1106 @@
+    }
+    return result;
+  }
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.SubdivideForChords`
+ * @internal
+ */
+class SubdivideForChordsSampler extends SubdivisionSampler {
+  private constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    super(c, o);
+  }
+  public static create(context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions): SubdivideForChordsSampler {
+    return new SubdivideForChordsSampler(context, options);
+  }
+  public override createErrorProcessor(): SubdivisionErrorProcessor {
+    return new LineStringSubdivisionErrorProcessor(this.fullArcXY, this._options.maxError);
+  }
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.SubdivideForArcs`
+ * @internal
+ */
+class SubdivideForArcsSampler extends SubdivisionSampler {
+  private constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    super(c, o);
+  }
+  public static create(context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions): SubdivideForArcsSampler {
+    return new SubdivideForArcsSampler(context, options);
+  }
+  public override createErrorProcessor(): SubdivisionErrorProcessor {
+    return new ArcChainSubdivisionErrorProcessor(this.fullArcXY, this._options.maxError);
+  }
+};
+
+/**
+ * Intermediate class for constructing a sample-based approximation.
+ * @internal
+ */
+class ConstructionProcessor extends QuadrantFractionsProcessor {
+  protected _chain: CurveChain;
+  protected _quadrantChain?: CurveChain;
+  public constructor(ellipticalArc: Arc3d) {
+    super();
+    this._chain = ellipticalArc.sweep.isFullCircle ? Loop.create() : Path.create();
+  }
+  public get chain(): CurveChain | undefined {
+    return this._chain.children.length > 0 ? this._chain : undefined;
+  }
+  public override announceQuadrantBegin(_q: QuadrantFractions): boolean {
+    this._quadrantChain = undefined;
+    return true;
+  }
+  public override announceQuadrantEnd(_q: QuadrantFractions, wasReversed: boolean): void {
+    if (this._quadrantChain) {
+      if (wasReversed)
+        this._quadrantChain.reverseChildrenInPlace();
+      for (const child of this._quadrantChain.children)
+        this._chain.tryAddChild(child); // captured!
+    }
+  }
+};
+
+/**
+ * Processor for constructing a sample-based arc chain approximation.
+ * @internal
+ */
+class ArcChainConstructionProcessor extends ConstructionProcessor {
+  public constructor(ellipticalArc: Arc3d) {
+    super(ellipticalArc);
+  }
+  public override announceArc(arc: Arc3d, _f0: number, _f1: number): void {
+    if (!this._quadrantChain)
+      this._quadrantChain = Path.create();
+    this._quadrantChain.tryAddChild(arc); // captured!
+  }
+};
+
+/**
+ * Processor for constructing a sample-based linestring approximation.
+ * @internal
+ */
+class LineStringConstructionProcessor extends ConstructionProcessor {
```

Comment:
`ChordConstructionProcessor` ?

Reply:
It returns a LineString3d, so LineString is more descriptive I think.